### PR TITLE
SIM: Add Rebalancing

### PR DESCRIPTION
**Pull request by BramGruneir**
This adds rebalancing to the simulator.

### CODE REVIEW

File: storage/simulation/main.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func main() {
 	c.flush()
 	fmt.Println(c)
 
-	// Split the last range 50 times.
+	// Split the last range 70 times.
```
Comment (tamird) [inline storage/simulation/main.go]: remove?

Comment (tamird) [inline storage/simulation/main.go]: derp, sorry. i see now that you haven't addressed comments yet.

Comment (BramGruneir) [inline storage/simulation/main.go]: all good, just pushed.  Was just marking comments as completed as I went through them.

**Comment by tamird** [comment]
LGTM after fixup

**Pull request merged**